src/WCSimLIGen.cc

```diff
@@ -0,0 +1,282 @@
+            }
+        }
+    
+    }	
+    
+    if ( photonMode ){
+        LoadPhotonList();
+    }
+    else {
+        LoadProfilePDF();
+    }
+}
+
+void WCSimLIGen::LoadPhotonList(){
+
+    string photonsFile = wcsimdir + photonsFilename;
+    cout << photonsFile << endl; 
+    ifstream photondata(photonsFile.c_str());
+    if ( ! photondata) {
+        cout << "WCSimLIGen [ERROR]: Failed to open " << photonsFilename << endl;
+    }
+    else {
+        cout << "WCSimLIGen photon list file: " << photonsFilename << " opened" << endl;
+    }
+
+    // Load in a csv list of x,y,z,px,py,pz in mm
+    string line;
+    // skip the first (header) line
+    getline(photondata,line);
+
+    while(getline(photondata,line)){
+        
+        Photon photon;
+        istringstream iss(line);
+        string value;
+        getline(iss,value, ',');
+        photon.x = stod(value);
+        getline(iss,value, ',');
+        photon.y = stod(value);
+        getline(iss,value, ',');
+        photon.z = stod(value);
+        getline(iss,value, ',');
+        photon.px = stod(value);
+        getline(iss,value, ',');
+        photon.py = stod(value);
+        getline(iss,value, ',');
+        photon.pz = stod(value);
+
+        myPhotons.push_back(photon);
+    }
+
+  photondata.close();
+
+}
+
+void WCSimLIGen::LoadProfilePDF(){
+
+    cout << "Reading the injector profile from file" << endl;
+
+    // Creates histogram of light injector profile
+    float thetaMin = thetaVals[0];
+    float phiMin = phiVals[0];
+    int nbins = thetaVals.size();
+    float thetaMax = thetaVals[nbins-1];
+    float phiMax = phiVals[nbins-1];
+
+    hProfile = new TH2D("hProfile","hProfile",nbins,thetaMin,thetaMax,nbins,phiMin,phiMax);
+    for (auto i=0u;i<thetaVals.size();i++){
+        hProfile->Fill(thetaVals[i],phiVals[i]);
+        int bin = hProfile->FindBin(thetaVals[i],phiVals[i]);
+        hProfile->SetBinContent(bin, intensity[i]);
+    }
+    cout << "Profile filled." << endl;
+}
+
+
+
+void WCSimLIGen::GeneratePhotons(G4Event* anEvent,G4int nphotons){
+
+    // Use particle gun to generate a point source
+    G4ParticleGun *myLIGun = new G4ParticleGun();
+    // Calculate photon energy now we have the wavelength
+    energy = PhotonEnergyFromWavelength(injectorWavelength);
+
+    if (photonMode){
+        // Get the position and direction from the photon list
+        // TODO subdivide into an equal number of photons per pulse 
+        // to limit the number of particles in G4Event
+        // for (int iphoton=0;iphoton<nphotons;iphoton++){
+        // have an index which is set in the constructor
+        // that way we can pick up the list where we left off last event
+    //    for (int iphoton=0;iphoton<myPhotons.size();iphoton++){
+        for (int iphoton=0;iphoton<nphotons;iphoton++){
+            // Generate random time for this photon in 20 ns pulse window
+            G4double time = G4RandFlat::shoot(20.0,40.0)*ns;
```

Comment:
Make sure to document the pulse width assumption more than just here (e.g. in the mac file)

Better may be to put this into the json? (naively I'd be suprised if the collimator/diffuser/ball have the same time profile)

Reply:
Following this up, it seems that at least for the diffuser and collimator the photon times are within 1 ns and on that basis there is no plan to measure the time profiles. So I have changed WCSimLIGen.cc to generate random times in a 1 ns window, and added a comment in the LIGen.mac. If the laser ball has a different time profile I will move it to the json at some point in the future.

Reply:
Sounds good to me